### PR TITLE
feat: add route for force crawling

### PR DESCRIPTION
**Pull request by bolinocroustibat**
Closes https://github.com/datagouv/data.gouv.fr/issues/1395 (partly).

This ticket has been split into 5 PRs for easier reviewing:
1) Refactoring DB methods for new routes: [PR #112](https://github.com/datagouv/hydra/pull/112) [✅](https://emojipedia.org/check-mark-button)
2) Refactoring of old routes [PR #117](https://github.com/datagouv/hydra/pull/117) [✅](https://emojipedia.org/check-mark-button)
3) Adding new route to gather informations from a resource in the catalog (ex : url, priority, etc.) [PR #120] [✅](https://emojipedia.org/check-mark-button)
4) **This PR**: Adding new route to force a crawl [PR #118](https://github.com/datagouv/hydra/pull/118)📍
5) Adding resource statuses and a resource status route: [PR #119](https://github.com/datagouv/hydra/pull/119)

**Details of this PR:**
- Create a new POST route `/api/checks/create/` to force a new crawl on a specific `resource_id` or `url`. _Note: no authentification in this PR yet, I think we should do it, but in another PR as this one is already big_
- Create a new related test `tests/test_api.py::test_api_resource_created`

### CODE REVIEW

File: tests/test_api.py
Comment (magopian) [inline tests/test_api.py:98]: The diff in github is a bit weird, but it makes it look like this line is gone? Is that on purpose?

**Review by magopian — APPROVED** [review]
Looks good to me :+1:

Do you want to add that to the CHANGELOG? (maybe we need a CI target that automatically checks the CHANGELOG has been updated... or even a tool that automatically populates the CHANGELOG from the commits ? To be discussed ;)

File: udata_hydra/routes/checks.py
Comment (Pierlou) [inline udata_hydra/routes/checks.py:37]: I find it weird to ask for both the resource's URL and id, and I guess it could be error-prone (what happens if you mismatch URL and id?). Wouldn't there be a way to manage with only the id? We could get the URL from the db or data.gouv directly 🤔

File: tests/test_api.py
Comment (Pierlou) [inline tests/test_api.py:113]: I'm guessing this will go eventually and was left as a reminder?

File: udata_hydra/routes/checks.py
Comment (Pierlou) [inline udata_hydra/routes/checks.py:50]: Wondering if in this specific case we'd like the check to have high priority. Currently `check_url` [always sets priority as low](https://github.com/datagouv/hydra/blob/46b73144792392557255a72f3ee63ed88bf2045f/udata_hydra/crawl.py#L288C27-L288C43). Thoughts?

**Comment by bolinocroustibat** [comment]
> Looks good to me 👍
> 
> Do you want to add that to the CHANGELOG? (maybe we need a CI target that automatically checks the CHANGELOG has been updated... or even a tool that automatically populates the CHANGELOG from the commits ? To be discussed ;)

Good point, I forgot. Changed now and also fixed a fe typos in the changelog.

I would be very interested by the second option, to populate the changelog automatically from commit names and descriptions. That would be a very big gain in productivity, and it would force us to do even better commits names and descriptions :)

**Review by Pierlou — APPROVED** [review]
Looking forwards to having this in production!

**Comment by bolinocroustibat** [comment]
I will wait for @maudetes to review the whole logic (especially the priority logic) to make sure it's ok, before merging.
So keeping this PR open for a few more days before merging :)

File: udata_hydra/cli.py
Comment (maudetes) [inline udata_hydra/cli.py:148]: Actually, since it's quite an equivalent from the API call, we may want to have `worker_priority="high"` as well in the cli.

File: udata_hydra/crawl.py
Comment (maudetes) [inline udata_hydra/crawl.py:245]: Thanks for the renaming, it is clearer indeed!

**Review by maudetes — APPROVED** [review]
Thank yo!

I am thinking of future iterations (running the different tasks inline instead of going through a worker, an additional priority level, etc.) but we can start with this version for a starter!

It would be wiser to test this new endpoint for real usage examples.

**Pull request merged**
